### PR TITLE
Reactive extension WKWebView navigation delegate

### PR DESCRIPTION
**Pull request by gringoireDM**
This PR add a new delegate proxy for `WKNavigationDelegate` for `WKWebView`.
Losing `UIWebView` delegate proxy, this PR now offers a replacement.

Please note that WKWebView has multiple delegates. This PR offers a proxy just for the `WKNavigationDelegate`. Other delegates will be subject of future PRs

### CODE REVIEW

File: Tests/RxCocoaTests/DelegateProxyTest.swift [hunk 1]
```diff
@@ -415,6 +415,7 @@ extension DelegateProxyTest {
 
             _ = (control as! NSObject).rx.deallocated.subscribe(onNext: { _ in
                 deallocated = true
+                expectation.fulfill()
```
Comment (gringoireDM) [inline Tests/RxCocoaTests/DelegateProxyTest.swift]: WKWebView is not immediately deallocated for some reason. We need to wait until this happens. The timeout on the `wait` will make sure that the test won't run forever

**Comment by ianloubser** [comment]
What's the plan with this PR ? 

Will this resolve the Appstoreconnect's deprecation warnings about UIWebView ?

**Comment by freak4pc** [comment]
UIWebView was already removed on 5.1.

@gringoireDM this needs a rebase on top of develop please

**Comment by gringoireDM** [comment]
@freak4pc just rebased

**Comment by gringoireDM** [comment]
@freak4pc is there anything else you want me to do?

**Comment by gringoireDM** [comment]
@freak4pc force pushing to develop makes continuous rebase really really hard...

**Comment by freak4pc** [comment]
Aware, very sorry. If you’re on the Rx slack it’s because I made some decisions on release schedules. It’s not something we usually do. Right now new additions will be put on top of RxSwift 5 and cherry picked to 6 as we go. There shouldn’t be extra changes here.

**Comment by gringoireDM** [comment]
👍 almost finished rebasing. If we could make normal commits I bet that most of PRs currently opened would benefits, but I also understand the importance of having a clean history on develop.

No worries, I'll keep the PR tidy.

**Comment by freak4pc** [comment]
We usually do, but the divergence I’d develop of 5.x vs develop 6.x is too vast

**Comment by gringoireDM** [comment]
history on this branch is gone... I'll close the PR and reopen it when we are ready to receive PRs again. Please let me know

**Comment by freak4pc** [comment]
Can you explain what do you mean ? Can’t you just cherry pick the last three commits on top of current 5.x develop?

**Comment by gringoireDM** [comment]
Sure I can, but there is no point if more changes are coming before this PR is ready to be merged. So I'll wait that the situation becomes "stable" :D

**Pull request closed without merging**